midp/content.js

```diff
@@ -0,0 +1,110 @@
+/* -*- Mode: JavaScript; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+'use strict';
+
+var Content = (function() {
+  var chRegisteredClassName = null;
+  var chRegisteredStorageID = -1;
+  var chRegisteredID = null;
+  var chRegisteredRegistrationMethod = -1;
+
+  function serializeString(parts) {
+    return parts.reduce(function(prev, current) {
+      return prev + String.fromCharCode(current.length * 2) + current;
+    }, "");
+  }
+
+  addUnimplementedNative("com/sun/j2me/content/RegistryStore.init.()Z", 1);
+
+  Native["com/sun/j2me/content/RegistryStore.forSuite0.(I)Ljava/lang/String;"] = function(suiteID) {
+    if (!chRegisteredClassName) {
+      return null;
+    }
+
+    var serializedString = serializeString([
+                                              chRegisteredID,
+                                              chRegisteredStorageID.toString(16),
+                                              chRegisteredClassName,
+                                              chRegisteredRegistrationMethod.toString(16)
+                                           ]);
+
+    return J2ME.newString(String.fromCharCode(serializedString.length * 2) + serializedString);
+  };
+
+  addUnimplementedNative("com/sun/j2me/content/RegistryStore.findHandler0.(Ljava/lang/String;ILjava/lang/String;)Ljava/lang/String;", null);
+
+  Native["com/sun/j2me/content/RegistryStore.register0.(ILjava/lang/String;Lcom/sun/j2me/content/ContentHandlerRegData;)Z"] = function(storageId, className, handlerData) {
+    chRegisteredID = J2ME.fromJavaString(handlerData.ID);
+    chRegisteredClassName = J2ME.fromJavaString(className);
+    chRegisteredStorageID = storageId;
+    chRegisteredRegistrationMethod = handlerData.registrationMethod;
+
+    return 1;
+  };
+
+  Native["com/sun/j2me/content/RegistryStore.getHandler0.(Ljava/lang/String;Ljava/lang/String;I)Ljava/lang/String;"] = function(callerId, ID, mode) {
```

Comment:
Nit: I'd capitalize the second parameter *id* for consistency with the declaration of this method in the RegistryStore class.